File: BinMedia/src/main/java/cn/zybwz/binmedia/bean/FilterInfo.java
package cn.zybwz.binmedia.bean;

import cn.zybwz.binmedia.UtilKt;

public class FilterInfo {
    private String showName;
    private String name;
    private String liveCmd;//
    private String fileCmd;
    private String[] args;
    private String fileIn;
    private String fileOut;

    public FilterInfo(String showName, String name, String liveCmd, String fileCmd, String[] args) {
        this.showName = showName;
        this.name = name;
        this.liveCmd = liveCmd;
        this.fileCmd = fileCmd;
        this.args = args;
    }

    public String getShowName() {
        return showName;
    }

    public void setShowName(String showName) {
        this.showName = showName;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getLiveCmd() {
        return liveCmd;
    }

    public void setLiveCmd(String liveCmd) {
        this.liveCmd = liveCmd;
    }

    public String getFileCmd() {
        return fileCmd;
    }

    public void setFileCmd(String fileCmd) {
        this.fileCmd = fileCmd;
    }

    public String[] getArgs() {
        return args;
    }

    public void setArgs(String[] args) {
        this.args = args;
    }

    public String getFileIn() {
        return fileIn;
    }

    public void setFileIn(String fileIn) {
        this.fileIn = fileIn;
    }

    public String getFileOut() {
        return fileOut;
    }

    public void setFileOut(String fileOut) {
        this.fileOut = fileOut;
    }

    public String buildLiveCmd(){
        return String.format(liveCmd, (Object[])args);
    }

    public String buildFileCmd(){
        String[] strings=new String[args.length+2];
        System.arraycopy(args,0,strings,1,args.length);
        strings[0]=fileIn;
        strings[strings.length-1]=fileOut;
        return String.format(fileCmd, (Object[])strings);
    }
}

File: BinMedia/src/main/java/cn/zybwz/binmedia/bean/FilterInfo.kt
package cn.zybwz.binmedia.bean

class FilterInfo {
    var showName:String=""
    var name:String=""
    var realTimeFormat=""
    var cmdFormat=""
    var defaultArg = arrayOf("")

    constructor(showName: String, name: String, realTimeFormat: String, cmdFormat: String) {
        this.showName = showName
        this.name = name
        this.realTimeFormat = realTimeFormat
        this.cmdFormat = cmdFormat
    }

    constructor(
        showName: String,
        name: String,
        realTimeFormat: String,
        cmdFormat: String,
        defaultArg: Array<String>
    ) {
        this.showName = showName
        this.name = name
        this.realTimeFormat = realTimeFormat
        this.cmdFormat = cmdFormat
        this.defaultArg = defaultArg
    }


}